source/_posts/2016/2016-12-24-how-to-avoid-inject-thanks-to-decorator-feature-in-nette.md

```diff
@@ -0,0 +1,254 @@
+---
+layout: post
+title: "How to avoid @inject thanks to Decorator feature in Nette"
+perex: "I often find <code>@inject</code> being overused in projects I review while mentoring. They often bring less writing, but in exchange they break SOLID principals. Today I will show you solution that will <strong>keep your code both small and clean</strong> - <strong>Decorator feature in Nette</strong>."
+lang: en
+---
+
+As [Derek Simons says](https://www.ted.com/talks/simon_sinek_how_great_leaders_inspire_action) says...
+
+## ...Start with "Why"
+
+Why am I writing this article? I try to improve knowledge interoperability between frameworks by finding the common lowest denominator, so it **is easier to understand and use each other**.
+
+I understand every framework has its own specific way how to solve same issues, but it creates *knowledge vendor lock*. So even if I know how to solve some problem in Nette, I cannot do that in Symfony and have to re-learn it again.
+ 
+Today, I will try to agree on setter injection with you. 
```

Comment:
This should be probably standalone article, hmm (todo).

Let me rephrase:

- magic injection is handled in every framework by different way (@inject, inject<>, Kdyby\Autowired in Nette x service locator, JMSDiExtra bundle in Symfony), different addons, bundles or extensions
- **yet they have the same goal**: get some dependency without using constructor injection
- I want to suggest to take common way, that is usable both in Symfony and Nette

Better?